server/protocols/webadmin/webadmin-mailrepository/src/test/java/org/apache/james/webadmin/service/ReprocessingServiceTest.java

```diff
@@ -0,0 +1,196 @@
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+
+import org.apache.james.filesystem.api.FileSystem;
+import org.apache.james.mailrepository.api.MailKey;
+import org.apache.james.mailrepository.api.MailRepository;
+import org.apache.james.mailrepository.api.MailRepositoryPath;
+import org.apache.james.mailrepository.api.MailRepositoryUrl;
+import org.apache.james.mailrepository.memory.MemoryMailRepositoryProvider;
+import org.apache.james.mailrepository.memory.MemoryMailRepositoryStore;
+import org.apache.james.mailrepository.memory.MemoryMailRepositoryUrlStore;
+import org.apache.james.queue.api.MailQueueFactory;
+import org.apache.james.queue.api.ManageableMailQueue;
+import org.apache.james.queue.api.RawMailQueueItemDecoratorFactory;
+import org.apache.james.queue.memory.MemoryMailQueueFactory;
+import org.apache.james.server.core.configuration.Configuration;
+import org.apache.james.server.core.configuration.FileConfigurationProvider;
+import org.apache.james.server.core.filesystem.FileSystemImpl;
+import org.apache.mailet.base.test.FakeMail;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.fge.lambdas.consumers.ConsumerChainer;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+
+public class ReprocessingServiceTest {
+    private static final String MEMORY_PROTOCOL = "memory";
+    private static final MailRepositoryPath PATH = MailRepositoryPath.from("path");
+    private static final String NAME_1 = "key-1";
+    private static final String NAME_2 = "key-2";
+    private static final String NAME_3 = "key-3";
+    private static final MailKey KEY_1 = new MailKey(NAME_1);
+    private static final MailKey KEY_2 = new MailKey(NAME_2);
+    private static final MailKey KEY_3 = new MailKey(NAME_3);
+    private static final String SPOOL = "spool";
+    private static final Consumer<MailKey> NOOP_CONSUMER = key -> {};
+    private static final Optional<String> NO_TARGET_PROCESSOR = Optional.empty();
+
+    private ReprocessingService reprocessingService;
+    private MemoryMailRepositoryStore mailRepositoryStore;
+    private MailQueueFactory<ManageableMailQueue> queueFactory;
+    private FakeMail mail1;
+    private FakeMail mail2;
+    private FakeMail mail3;
+
+    @Before
```

Comment:
junit 5?

Reply:
I want to keep this bugfix changeset small and uncouple it from huge refactorings.

Reply:
nice :) just for the record, this PR is already +469 −31. You could already have split it, handling multi  repo in a different PR than mono repo issues.

Reply:
But lead to very similar fix, hence my choice.

IMO it did not pass the "show-stopper point"